spmd/tensor/redistribute.py

```diff
@@ -158,14 +157,34 @@ def redistribute_spmd_tensor(
                     new_local_tensor = local_tensor
             else:
                 raise RuntimeError(
-                    f"redistribute from {current_placements} to {placements} not supported yet"
+                    f"redistribute from {current_placements} to {target_placements} not supported yet"
                 )
 
         assert new_local_tensor is not None
         local_tensor = new_local_tensor
 
     assert new_local_tensor is not None, "redistribute failed!"
 
+    return new_local_tensor
+
+
+def redistribute_spmd_tensor(
```

Comment:
can we merge this with the `Redistribute` autograd function?

Reply:
In that case, I assume `pack_args_kwargs_with_local_tensor` will then also need to call into `Redistribute` autograd function, which occurs in the `__torch_dispatch__` function under `no_grad` mode?  

https://github.com/pytorch/tau/blob/c71e2866015ca23beee4e17c9d8dae415d5f86b4/spmd/tensor/utils.py#L57

Reply:
Oh I mean we should probably also change `pack_args_kwargs_with_local_tensor` by calling into `_redistribute_with_local_tensor` directly? so that we could safely delete this `redistribute_with_spmd_tensor` api

Reply:
sure, let me update that.

Reply:
Hey @wanchaol , I tried update `pack_args_kwargs_with_local_tensor` and remove `redistributed_dtensor`, but code becomes a bit verbose, as `_redistribute_with_local_tensor` take more argument and requires one additional DTensor wrapping. Let me know if you prefer to get rid of `redistributed_dtensor`. I can do that in a follow up PR.